usd_for_web_demos/test.html

```diff
@@ -38,9 +38,16 @@
 
     Promise.all([loadUSDPromise, getUsdModule(), initPromise]).then(async ([usdFile, Usd]) => {
       window.Usd = Usd;
+      loadUsdFileFromArrayBuffer(filename, usdFile);
+    });
 
-      let extension = filename.split('.')[1];
-      let inputFile = 'input.' + extension;
+    function loadUsdFileFromArrayBuffer(filename, usdFile) {
+      let parts = filename.split('.');
+      let extension = parts[parts.length - 1];
+      extension = extension.split('?')[0];
+      let fileName = (Math.random() + 1).toString(36).substring(7);
+      // fileName = "input"; // TODO need to unload the old file again, then we don't need a random filename here
+      let inputFile = fileName + "." + extension;

```

Comment:
This is using a random filename because otherwise the file already exists on subsequent loads.
The "old" file needs to be removed here, but I'm not sure how.

Reply:
@kaischroeder would be looking for your input here especially - unloading the previous file is the main part still missing to make this useful.

Reply:
is this just about the data file? In this case you could try unlinking it - this is using the Emscripten Filesystem API:

``Usd.FS.unlink('/' + inputFile);``


https://emscripten.org/docs/api_reference/Filesystem-API.html#FS.unlink

The only question is when :-) Doing it after calling driver.Draw() should be safe I guess.

Reply:
I tried to unlink actually, but then got all kinds of exceptions from what I assume to be residuals of registering the previous file with .Draw. I didn't find a way to "unregister" it again.
Without that, both the old and new file will be rendered...

Reply:
Got it to work! Key was to not only unlink the file but also remove the created objects from the Three.js scene graph.